### ISSUE TITLE
Feature request: linting workflow

### ISSUE THREAD

**Issue by tryonkus** [otaviocc/obsidian-microblog#22]
I have recently started using micro.publish to publish to micro.blog more often, and I ran into an issue I'm hoping you could help with. 
* I use a plugin to pull scripture quotations into blockquotes, although this could happen with vanilla Obsidian.
* The plugin labels the blockquotes (Obsidian calls these [callouts](https://help.obsidian.md/Editing+and+formatting/Callouts)).
* Micro.blog supports block quotes but not callouts—they are supported in a newer version of Hugo, but I don't know when that will be implemented (I've been on the micro.blog forum about this as well).

One solution for me would be to simply run a linter or regex to remove the labels from blockquotes before posting—I use both the Linter and Regex Pipeline plugins. If there was a way I could trigger either of those automatically as part of my publishing workflow, that would be boffo.

I may already be able to do this with the Commander plugin or anything that lets me define Obsidian macros or workflows—I may even be able to do it with Templater, since that lets me run Javascript code. (I'm in IT, but I know only rudimentary JavaScript.)

Judging from your blog, this has been an eventful year, and Obsidian didn't make it into your December 2024 defaults. If you could find time to do this, I'd buy you several cups of coffee. ☕

**Comment by tryonkus** [comment]
Update: I learned that the Commander plugin supports simple macros, and I created one to do the following:
* Run a series of regex replacements via the Regex Rulesets plugin
     * Remove level 1 header (I use the post/page title instead)
     * Remove %% comments
     * Remove labels on blockquotes/callouts (these are supported in the most recent version of Hugo, but I haven't tweaked my post styling to take advantage of that yet).
 * pause one second
 * Publish post via micro.publish
 * pause one second
 * Undo edits (one undo restores everything removed in the first step)

I discovered that the undo step worked fine when I ran the process manually, but it didn't work properly in the macro. I added the pauses, and that fixed the problem—likely the publish and undo steps were colliding. I did notice that the undo step executes while I'm still in the micro.publish dialogue, but the post still publishes correctly. The plugin must dump the text into a buffer which isn't affected by the undo command.

I'd still love to have the ability to preprocess text before publishing, but using Commander and Regex Rulesets allows me to do what I need and probably gives me more flexibility than preprocessing native to the micro.publish plugin would.
